pkg/comp-functions/functions/vshnforgejo/deploy.go

```diff
@@ -0,0 +1,204 @@
+
+	err := svc.GetObservedComposite(comp)
+	if err != nil {
+		return runtime.NewFatalResult(fmt.Errorf("cannot get composite: %w", err))
+	}
+
+	svc.Log.Info("Bootstrapping instance namespace and rbac rules!")
+	err = common.BootstrapInstanceNs(ctx, comp, comp.GetServiceName(), comp.GetName()+"-instanceNs", svc)
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot bootstrap instance namespace: %s", err))
+	}
+
+	secretName, err := common.AddCredentialsSecret(comp, svc, []string{"password"}, common.AddStaticFieldToSecret(map[string]string{
+		"username": "gitea_admin",
+	}))
+	if err != nil {
+		return runtime.NewWarningResult("cannot add credentials secret")
+	}
+
+	connDetails, err := svc.GetObservedComposedResourceConnectionDetails(secretName)
+	if err != nil {
+		return runtime.NewWarningResult("cannot get connection details")
+	}
+
+	svc.SetConnectionDetail("FORGEJO_USERNAME", connDetails["username"])
+	svc.SetConnectionDetail("FORGEJO_PASSWORD", connDetails["password"])
+	// "," as separator for multiple FQDNs, it's better than anything else as highlighting with mouse in terminal works well
+	// result is fqdn[0],fqdn[1],fqdn[2]
+	svc.SetConnectionDetail("FORGEJO_URL", []byte(strings.Join(comp.Spec.Parameters.Service.FQDN, ",")))
+
+	svc.Log.Info("Adding forgejo release")
+	err = addForgejo(ctx, svc, comp, secretName)
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot add forgejo release: %s", err))
+	}
+
+	return nil
+}
+
+func addForgejo(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VSHNForgejo, secretName string) error {
+
+	securityContext := map[string]any{}
+	if svc.Config.Data["isOpenshift"] == "true" {
+		securityContext = map[string]any{
+			"runAsUser":                nil,
+			"allowPrivilegeEscalation": false,
+			"podSecurityContext": map[string]any{
+				"fsGroup": nil,
+			},
+			"containerSecurityContext": map[string]any{
+				"runAsUser":                nil,
+				"allowPrivilegeEscalation": false,
+			},
+			"capabilities": map[string]any{
+				"drop": []string{
+					"ALL",
+				},
+			},
+		}
+	}
+
+	values := map[string]any{
+		"gitea": map[string]any{
+			"admin": map[string]any{
+				"existingSecret": secretName,
+			},
+			"config": map[string]any{
+				"admin": map[string]any{
+					"SEND_NOTIFICATION_EMAIL_ON_NEW_USER": true,
+				},
+				"cache": map[string]any{
+					"ADAPTER": "twoqueue",
+					"HOST":    "'{\"size\":100, \"recent_ratio\":0.25, \"ghost_ratio\":0.5}'",
+				},
+				"cron": map[string]any{
+					"ENABLED": true,
+					"archive_cleanup": map[string]any{
+						"SCHEDULE":   "@hourly",
+						"OLDER_THAN": "2h",
+					},
+				},
+				"database": map[string]any{
+					"DB_TYPE": "sqlite3",
+				},
+				"image": map[string]any{
+					"tag": comp.Spec.Parameters.Service.Version,
+				},
+				"indexer": map[string]any{
+					"ISSUE_INDEXER_TYPE":   "bleve",
```

Comment:
We should probably document somewhere that we enable this by default. The bleve search index will be the same size as all the code.

Reply:
This is already in our arhcitecture document, but I'll add information that REPO indexing consumes 6 times more disk space than repo itself.

https://docs.gitea.com/administration/config-cheat-sheet#indexer-indexer